<launch>
    <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">

        <rosparam file="$(find amcn)/configs/costmap_common_params.yaml" command="load" ns="global_costmap" />
        <rosparam file="$(find amcn)/configs/costmap_common_params.yaml" command="load" ns="local_costmap" />
        <rosparam file="$(find amcn)/configs/odom_nav_params/global_costmap_params.yaml" command="load" />
        <rosparam file="$(find amcn)/configs/odom_nav_params/local_costmap_params.yaml" command="load" />

        <param name="base_global_planner" type="string" value="navfn/NavfnROS" />
        <param name="base_local_planner" value="base_local_planner/TrajectoryPlannerROS" />

        <rosparam file="$(find amcn)/configs/base_local_planner_params.yaml" command="load" />
        <rosparam file="$(find amcn)/configs/move_base_params.yaml" command="load" />

        <remap from="odom" to="odometry/filtered" />
        <remap from="cmd_vel" to="cmd_vel_faked_by_move_base" />
    </node>

    <node pkg="amcn" type="mpc_node.py" name="mpc_controller" output="screen">
        <remap from="global_plan" to="/move_base/NavfnROS/plan" />
        <remap from="cmd_vel" to="/cmd_vel" />

        <rosparam file="$(find amcn)/configs/mpc_params.yaml" command="load" />
    </node>

    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find amcn)/rviz/move_base.rviz" output="screen"/>

</launch>
